.github/workflows/smoketest.yml

```diff
@@ -0,0 +1,95 @@
+on:
+  pull_request:
+  merge_group:
+name: SmokeTest
+concurrency:
+  group: ${{ github.ref }}-smoketest
+  cancel-in-progress: true
+jobs:
+  smoke-test-upgrade-path:
+    name: Smoke Test Upgrade Path
+    # if: github.event_name != 'pull_request' || github.event.action == 'enqueued' || contains( github.event.pull_request.labels.*.name, 'run-all')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository with full history
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Get last tagged release
+        run: |
+          latest_release=$(git tag --sort=-v:refname | grep -v v1 | head -n 1)
```

Comment:
At some point (not now) we are going to need to think about testing older releases. The 'latest release' is often just a few commits ago.